pkg/controller/projects/projects/project.go

```diff
@@ -149,6 +158,52 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errNotProject)
 	}
+	// Line 161 -> 206 will be export to an util function
+	keySecretRef := cr.Spec.ForProvider.ImportUrlSecretRef
+	importUrl := cr.Spec.ForProvider.ImportURL
+
+	// User has provided secret for the import
+	if keySecretRef != nil && importUrl != nil {
+		// Retrieve secret from k8s
+		namespacedName := types.NamespacedName{
+			Namespace: keySecretRef.Namespace,
+			Name:      keySecretRef.Name,
+		}
+
+		secret := &corev1.Secret{}
+
+		err := e.kube.Get(ctx, namespacedName, secret)
+
+		if err != nil {
+			return managed.ExternalCreation{}, errors.New(errNoImportSecret)
+		}
+
+		// Obtain the password from the secret.
+		password := string(secret.Data[keySecretRef.PasswordKey])
+		if password == "" {
+			return managed.ExternalCreation{}, errors.New(errPasswordNotFound)
+		}
+
+		// Obtain the username from the secret.
+		username := string(secret.Data[keySecretRef.UsernameKey])
+		if username == "" {
+			return managed.ExternalCreation{}, errors.New(errUsernameNotFound)
+		}
+
+		// manipulate url to add the secret. If secret is already in the url, it should be overridden
```

Comment:
We should add this descrption also to the CRD property description, so it is clear that already provided credentials in an encoded URL will be overwritten in case a valid secretRef is present as well.